### ISSUE TITLE
tweakGammaOverride does not work when entering the world

### ISSUE THREAD

**Issue by SolidBlock-cn** [maruohon/tweakeroo#348]
When initially entering the world, gamma is as vanilla even if tweakGammaOverride is true. It takes effect only when disabled and then re-enabled.

**Comment by end-user** [comment]
I also started experiencing this, but only when I just recently upgraded to 1.19.2 (tweakeroo v0.14.3, malilib v0.13.0).

**Comment by ghost** [comment]
Same. I've noticed this to as well.

**Comment by aditeya24** [comment]
It's a known bug with the way the mod stores and/or restores (or doesn't) the gamma value after game start. It just doesn't apply properly when the game starts.

**Comment by end-user** [comment]
Ok, well, the bug didn't start showing up for me until this most recent update.  Is there an official bug this should be linked to?

**Comment by maruohon** [comment]
I *think* it was always broken/not working properly, but the 1.19 community version had a workaround/hack fix for it, but my official 1.19+ versions again don't have that fix, as I want to add a better fix for later on.

**Comment by end-user** [comment]
Ooh.  Yeah, that's likely.

**Comment by ghost** [comment]
Would it be more worth it just disabling the file that resets the gamma level? is that possible or is there more then that?

**Comment by muluffel** [comment]
Yup, I can confirm that the gamma override needs no be reenabled every time when entering a world or server

**Comment by DarkCows** [comment]
I forked the mod and fixed this issue: https://github.com/DarkCows/tweakfork-fork
Whenever the world loads I check if the gamma override setting is true and change the MC gamma setting.

**Comment by Phoupraw** [comment]
Same.
Why is https://github.com/maruohon/tweakeroo/pull/434 closed?

**Comment by wendavid552** [comment]
Goes still for latest stable version 1.20.4-0.19.1.

**Comment by axialeaa** [comment]
This is intentional as far as vanilla minecraft. The fix of MC-51418 prevented values outside of the normal slider ranges to be saved by the client. There's a mod by apple502j that "unpatches" this called FiveOneFourOneEight, but it hasn't been updated to accept newer versions in fabric.mod.json yet.

**Comment by maruohon** [comment]
Tweakeroo does not rely on the vanilla value being saved, it's *supposed to* always re-apply the value after the game launches, but that code has just been buggy/broken/using bad logic for ages. The fix has just kinda been waiting for my mod rewrites... (although the fix isn't large or difficult by any means ofc, I just don't put any effort to the old code mod versions anymore...)
